crates/codegen/parser/runtime/src/cursor.rs

```diff
@@ -133,6 +134,16 @@ impl Cursor {
         self.current.node.clone()
     }
 
+    pub fn node_name(&self) -> String {
```

Comment:
Same suggestion here about using `Option<String>` for unnamed nodes.

Reply:
I understand we are going to eventually use an `enum` to store all possible names, but that would not be exposed externally.
Users would still get `Option<String>` in the public API, right? or to be more accurate, `Option<&'static str>`.

Reply:
See https://github.com/NomicFoundation/slang/pull/710#discussion_r1427789453 specifically for the Option suggestion.

About not exposing the enum - don't we plan to do that? Otherwise users would have to encode the value set themselves/check the values of the strings rather than being able to refer to them using the constrained domain of the enum.

Reply:
mmm, I didn't think about exposing it publicly yet, but I guess it won't hurt to do it now. Sounds good to me!